catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -811,20 +811,25 @@ private void initScreenMode() {
 		camera.update();
 	}
 
-	private void disposeTextures() {
-		List<Sprite> sprites = scene.getSpriteList();
-		int spriteSize = sprites.size();
-		for (int i = 0; i > spriteSize; i++) {
-			List<LookData> data = sprites.get(i).getLookDataList();
-			int dataSize = data.size();
-			for (int j = 0; j < dataSize; j++) {
-				LookData lookData = data.get(j);
-				lookData.getPixmap().dispose();
-				lookData.getTextureRegion().getTexture().dispose();
+	private void disposeScenes() {
+		for (Scene scene : ProjectManager.getInstance().getCurrentProject().getSceneList()) {
+			List<Sprite> sprites = scene.getSpriteList();
+			for (Sprite sprite : sprites) {
+				List<LookData> data = sprite.getLookDataList();
+				disposeTextures(data);
```

Comment:
* `disposeScenes` isn't a good name. Overload `disposeTextures` to take a Project object instead.
* Inline `scene.getSpriteList()`
* `ProjectManager.getInstance().getCurrentProject()` => `project`